executor/executor_test.go

```diff
@@ -1850,6 +1850,12 @@ func (s *testSuiteP1) TestGeneratedColumnRead(c *C) {
 	result = tk.MustQuery(`SELECT * FROM test_gc_read_m`)
 	result.Check(testkit.Rows(`1 2 4`, `2 3 6`))
 
+	// Test for issue #12233
+	tk.MustExec(`create table testjson( id int auto_increment not null primary key, j json )default charset=utf8 engine=innodb;`)
+	tk.MustExec(`insert into testjson set j='{"test":1}';`)
+	result = tk.MustQuery(`select id from testjson where json_extract(j, '$.test');`)
+	result.Check(testkit.Rows(`1`))
```

Comment:
I've tried to checkout your pull request, build it and run in a local environment, and it returns an error.

```mysql
mysql> select id from testjson where json_extract(j, '$.test');
ERROR 1105 (HY000): InvalidDataType("can\'t convert Json(1) to bool")
```

Could you do an investigation why it works well in unit test but fails in a local environment?

Reply:
In my local environment, it works well. Did you checkout this PR? Or describe your test steps in detail.
![image](https://user-images.githubusercontent.com/13314548/66803524-6c02f280-ef52-11e9-9309-df57875c4c43.png)

Reply:
It's my fault.